game.py

```diff
@@ -223,17 +196,15 @@ def perform_self_destruct(self, coords: CoordPair) -> Tuple[bool, str]:
     def perform_move(self, coords: CoordPair) -> Tuple[bool, str]:
         """Validate and perform a move expressed as a CoordPair. TODO: WRITE MISSING CODE!!!"""
         if self.is_valid_move(coords):
-            if self.is_movement(coords):
+            if self.is_empty(coords.dst):
                 return self.perform_movement(coords)
-            elif self.is_attack(coords):
+            elif self.next_player != self.get(coords.dst).player:
                 return self.perform_attack(coords)
-            elif self.is_self_destruct(coords):
+            elif self.is_self_move(coords):
                 return self.perform_self_destruct(coords)
-            elif self.is_repair(coords):
-                return self.perform_repair(coords)
             else:
```

Comment:
for perform_repair instead of it going into in an else it should have its own elif condition like i had done before in case there is a bug or we change the code down the line the default behavior will be is repair and itll cause a lot more problems
its also bad practice one move should not be a default instead the else should return an invalid move